File: VotingSystem.js
const prompt = require("prompt-sync")();
let candidates = [];
let voters = [];
let votes = {};
let votedVoters = [];

function registerCandidate(){
	let name = prompt(" Enter candidate Name:");
	let age = parseInt(prompt(" Enter candidate Age:"));
	let party = prompt(" Enter candidate political party:");
	let relatioshipStatus = prompt(" Enter candidate relatioship status:").toLowerCase();


	if(relatioshipStatus == "married" && age >= 18){
		candidates.push(name);
		votes[name] = 0;
		console.log(name, "has been successfuly registered");
	} else {
		console.log("candidate doesn't meet the requirement");
		}
				
}

function registerVoters(){
	let name = prompt(" Enter your Name:");
	let age = parseInt(prompt(" Enter your Age:"));
	let nationality = prompt("Are you a nigerian:").toLowerCase();
	if(age >= 18 && nationality == "yes"){
		voters.push(name);
		console.log(name, "has been successfuly registered and eligible to vote");
	} else {
		console.log("you are not eligible");
		}
}


function toVote(){
	let name = prompt(" Enter your Name:");
	if (voters.includes(name) && !votedVoters.includes(name)) {
		console.log("Candidates:");
        candidates.forEach(candidate => console.log(candidate));
        let choice = prompt("Enter the name of the candidate you want to vote for:");

	if (candidates.includes(choice)) {
            votes[choice]++;
            votedVoters.push(name);
            console.log("Vote cast successfully");
        } else {
            console.log("Invalid candidate");
        }
    } else if (votedVoters.includes(name)) {
        console.log("You have already voted.");
    } else {
        console.log("You are not a registered voter.");
    }
}

function countVotes() {
    
    for (let candidate in votes) {
        console.log(`${candidate}: ${votes[candidate]} vote(s)`);
    }
    console.log();
}




function main(){
	let hunterly = true
	while(hunterly){
		console.log("Voting System Menu");
		console.log("1.Candidate Resgistration");
		console.log("2.Voters Resgistration");
		console.log("3.Voting");
		console.log("4.count votes");
		console.log("0.Exit");

	let option = parseInt(prompt("enter an option from (1-6):"));
	switch(option){
		case 1:
			console.log("Candidate Registration");
			registerCandidate();
			break;
		case 2:
			console.log("Voters Registration");
			registerVoters();
			break;
		case 3:
			console.log("Voting");
			toVote();
			break;
		case 4:
			console.log("Counting Votes");
			countVotes();
			break;
		case 0:
			console.log("bye bye ");
			hunterly = false;
			break;
		default:
			console.log("invalid option. Try again");


}
}
}
main();

File: voting_system.py
candidates = []
voters = []
votes = {}
voted_voters = []


def register_candidate():
	name = input(" Enter candidate Name:");
	age = int(input(" Enter candidate Age:"));
	party = input(" Enter candidate political party:");
	relatioshipStatus = input(" Enter candidate relatioship status:").lower();


	if(relatioshipStatus == "married" and age >= 18):
		candidates.append(name);
		votes[name] = 0;
		print(name + " has been successfuly registered");
	else:
		print("candidate doesn't meet the requirement");



def register_voters():
	name = input(" Enter your Name:");
	age = int(input(" Enter your Age:"));
	nationality = input("Are you a nigerian:").lower();
	if(age >= 18 and nationality == "yes"):
		voters.append(name);
		print(name + " has been successfuly registered and eligible to vote");
	else:
		print("you are not eligible");


def vote_process():
    print("Voting ")
    name = input("Enter your name: ")
    if name in voters and name not in voted_voters:
        print("Candidates:")
        for candidate in candidates:
            print(f" {candidate}")
        choice = input("Enter the name of the candidate you want to vote for: ")
        if choice in candidates:
            votes[choice] += 1
            voted_voters.append(name);
            print("Vote cast successfully");
        else:
            print("Invalid candidate.");
    elif name in voted_voters:
        print("you have already voted");
    else:
        print("You are not a registered voter");
 
def count_votes():
    print("Counting Votes ")
    for candidate, count in votes.items():
        print(f"{candidate}: {count} vote(s)")
    print()


def main():
	while True:
		print(" Voting System Menu");
		print("1. Register Candidate");
		print("2. Register Voter");
		print("3. Vote");
		print("4. Count Votes");
		print("0. Exit");
		option = int(input("Select any option: "));
		match(option):
			case 1:
				print("Register Candidate");
				register_candidate();
			case 2:
				print("Register Voter");
				register_voters();
			case 3:
				print("Vote");
				vote_process();
			case 4:
				print("Count Votes");
				count_votes();
			case 0:
				print("bye bye");
				break;
			case _:
				print("Invalid choice. Try again");

main()


